### PR TITLE
Add hedgey-multi strategy [hedgey-multi]

### PR DESCRIPTION
**Pull request by bark4mark**
Creates a new strategy for the Hedgey protocol
Changes proposed in this pull request:
 - Adds a function to check multiple hedgey contracts
 - A multiplier function to manipulate the scores based on the deal parameters

Hedgey: https://hedgey.finance

### CODE REVIEW

File: src/strategies/hedgey-multi/index.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+export async function strategy(
+  _space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const contractDetails: ContractDetails[] = options.contracts;
```
Comment (ChaituVR) [inline src/strategies/hedgey-multi/index.ts:46]: Can we add a limit to `contracts` options? someone passes in unlimited contracts. better to throw error if someone pass more contract

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/hedgey-multi/index.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+    const amount = BigNumber.from(deal.amount).div(
+      BigNumber.from(10).pow(contractDetail.decimal)
+    );
+
+    let score = amount.toNumber();
+
+    if (contractDetail.multiplyer) {
+      const multiplyer = eval(contractDetail.multiplyer);
+      score = multiplyer(score, deal);
```
Comment (ChaituVR) [inline src/strategies/hedgey-multi/index.ts:150]: Hey @bark4mark Sorry for not noticing this earlier, But we can accept `eval` , There is a chance of security risk. can you try passing the multiplier value directly?

File: src/strategies/hedgey-multi/index.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    if (!compareAddresses(contractToken, contractDetail.token)) continue;
+
+    const amount = BigNumber.from(deal.amount).div(
+      BigNumber.from(10).pow(contractDetail.decimal)
+    );
+
+    let score = amount.toNumber();
+
+    if (contractDetail.multiplyer) {
+      const multiplyer = eval(contractDetail.multiplyer);
```
Comment (ChaituVR) [inline src/strategies/hedgey-multi/index.ts:179]: same here

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/hedgey-multi/index.ts (hunk 2 above)
Comment (bark4mark) [reply src/strategies/hedgey-multi/index.ts]: I have removed the eval

File: src/strategies/hedgey-multi/index.ts (hunk 3 above)
Comment (bark4mark) [reply src/strategies/hedgey-multi/index.ts]: I have removed the eval

**Comment by bark4mark** [comment]
@ChaituVR I have made the requested changes, thank you.

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**